CRM/Core/BAO/Dashboard.php

```diff
@@ -357,11 +357,30 @@ public static function saveDashletChanges($columns, $contactID = NULL) {
       }
     }
 
-    // Disable inactive widgets
+    // Find dashlets in this domain.
+    $domainDashlets = civicrm_api3('Dashboard', 'get', [
+      'return' => array('id'),
+      'domain_id' => CRM_Core_Config::domainID(),
+    ]);
+
+    // Get the array of IDs.
+    $domainDashletIDs = [];
+    if ($domainDashlets['is_error'] == 0) {
+      foreach ($domainDashlets['values'] as $domainDashlet) {
+        $domainDashletIDs[] = $domainDashlet['id'];
+      }
+    }
+
+    // Restrict query to Dashlets in this domain.
+    $domainDashletClause = !empty($domainDashletIDs) ? "dashboard_id IN (" . implode(',', $domainDashletIDs) . ")" : '(1)';
```

Comment:
is it possible to paramatise this query at all?

Reply:
@seamuslee001 I'm not clear how to do this - I've always assumed that passing params to `executeQuery` is for validating... _ahem_... params_ rather than building queries. I'm happy to add a commit which passes `contact_id` in as 'Integer' if that's what you meant.

Reply:
@seamuslee001 Commit c0543d0 passes the contact ID as a parameter. Hopefully this resolves this question.